src/web/components/ReaderRevenueLinks.test.tsx

```diff
@@ -17,8 +17,6 @@ jest.mock('@frontend/web/browser/cookie', () => ({
 describe('ReaderRevenueLinks', () => {
```

Comment:
Within this test suite we had 3 tests, corresponding to the logic in ReaderRevenueLinks source, but we've dramatically changed that logic now. It no longer checks several conditions, the logic for which is within that file. 

Instead we're just making one call out to `shouldHideSupportMessaging` and rendering based off that, so I think the tests should be updated to match.  My suggestion would be 2 tests, one that checks nothing is rendered when `true` is returned, and the other that checks _something_ is rendered when `false` is returned.

To control the return value of `shouldHideSupportMessaging` you can mock it out, as has been done for `getCookie`/`addCookie` previously. You can also remove the mocking for those things, because they're no longer needed.

Last thing to check (if being completely thorough) would be to see if the tests we're removing here have already been implemented in the contributions utility we're now importing. If not, then we ought to add some, or add a card to the backlog to do that later (or more likely ask another team to own this).

Reply:
Ok, I'll update the test cases. 

I guess it would be best to ask the reader revenue team to update the contributions, it needs to have digital subscriber checks added to it as well.

Reply:
I think digital subscriber would be included in the `gu_hide_support_messaging` cookie that already gets used.  (In fact all relevant products are). I suspect the extra checks for contributions are because they had an extra requirement to immediately remove all support messaging after a contribution - I don't think that necessarily makes sense for subscriptions.